ci/build.yml

```diff
@@ -6,7 +6,7 @@ image_resource:
   type: docker-image
   source:
     repository: golang
-    tag: 1.20.4
+    tag: 1.21
```

Comment:
This needs to be a [specific version](https://github.com/golangci/golangci-lint/releases) and your totally doing the right thing but I seem to remember reading somewhere our builds are going a bit flaky after `1.20.4` but can't find the thread now 😢 so I'd leave the version as is